modules/dnn/src/onnx/onnx_importer.cpp

```diff
@@ -0,0 +1,582 @@
+            layerParams.set("bias_term", node_proto.input_size() == 3);
+        }
+        else if (layer_type == "MatMul")
+        {
+            CV_Assert( node_proto.input_size() == 2);
+            layerParams.type = "InnerProduct";
+            Mat blob = getBlob(node_proto, constBlobs, 1);
+            layerParams.blobs.push_back(blob.t());
+            layerParams.set("bias_term", false);
+            layerParams.set("num_output", layerParams.blobs[0].size[0]);
+        }
+        else if (layer_type == "Mul")
+        {
+            CV_Assert(node_proto.input_size() == 2);
+            if (layer_id.find(node_proto.input(1)) == layer_id.end()) {
+                Mat blob = getBlob(node_proto, constBlobs, 1);
+                if (blob.total() == 1) {
+                    layerParams.set("scale", blob.at<float>(0));
+                    layerParams.type = "Power";
+                }
+                else {
+                    layerParams.blobs.push_back(blob);
+                    layerParams.type = "Scale";
+                }
+            }
+            else {
+                layerParams.type = "Eltwise";
+                layerParams.set("operation", "prod");
+            }
+        }
+        else if (layer_type == "Conv")
+        {
+            layerParams.type = "Convolution";
+            for (int j = 1; j < node_proto.input_size(); j++) {
+                layerParams.blobs.push_back( getBlob(node_proto, constBlobs, j) );
+            }
+            layerParams.set("num_output", layerParams.blobs[0].size[0]);
+            layerParams.set("bias_term", node_proto.input_size() == 3);
+        }
+        else if (layer_type == "Unsqueeze")
+        {
+            Mat input = getBlob(node_proto, constBlobs, 0);
+            DictValue axes = layerParams.get("axes");
+            std::vector<int> dims(input.dims + axes.size(), 0);
+            for (int j = 0; j < axes.size(); j++) {
+                CV_Assert(layerParams.get("axes").getIntValue(j) < dims.size());
+                dims[layerParams.get("axes").getIntValue(j)] = 1;
+            }
+            std::vector<int>::iterator it;
+            for (int j = 0; j < input.dims; j++) {
+                it = find(dims.begin(), dims.end(), 0);
+                CV_Assert(it != dims.end());
+                *it = input.size[j];
+            }
+            Mat out = input.reshape(0, dims);
+            constBlobs.insert(std::make_pair(node_proto.output(0), out));
+            continue;
+        }
+        else if (layer_type == "Reshape")
+        {
+            Mat blob = getBlob(node_proto, constBlobs, 1);
+            CV_Assert(blob.type() == CV_32SC1);
+            layerParams.set("dim", DictValue::arrayInt<int*>(
+                        blob.ptr<int>(), blob.total() ));
+
+            if (layer_id.find(node_proto.input(0)) == layer_id.end()) {
+                Mat input = getBlob(node_proto, constBlobs, 0);
+                std::vector<int> dims;
```

Comment:
Try to cast a mat with shapes to std::vector:
```cpp
std::vector<int> dims = blob;
```

Reply:
done